{{cookiecutter.project_slug}}/.github/workflows/main.yaml

```
@@ -6,20 +6,52 @@ jobs:
   check-lint-and-formatting:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
       - name: Check lint and formatting
         run: make check-lint-and-formatting
 
-  {% if cookiecutter.create_react_frontend == 'y' %}
+{% if cookiecutter.create_react_frontend == 'y' %}
   check-lint-and-test-frontend:
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v4
       - uses: actions/setup-node@v3
         with:
           node-version: '16'
       - name: Frontend Lint & Typecheck & Test
         run: make check-lint-and-test-frontend
-  {% endif %}
\ No newline at end of file
+{% endif %}
+
+  backend-test:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_DB: my_awesome_sixie_project
+          POSTGRES_USER: my_awesome_sixie_project
+          POSTGRES_PASSWORD: TESTPASSWORD
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Run backend tests
+        run: make backend-test
+        env:
+          DATABASE_URL: postgres://my_awesome_sixie_project:TESTPASSWORD@127.0.0.1:5432/my_awesome_sixie_project
+          POSTGRES_HOST: localhost
```

Use the hostname of the postgres service in the Kubernetes cluster:
```suggestion
          POSTGRES_HOST: postgres
```